### PR TITLE
Export multiBalanceReport helper functions and generalise compoundBalanceReport.

### PR DESCRIPTION
**Pull request by Xitian9**
This exports a few helper functions and generalises compoundBalanceReport to make them a bit more useful for writing your own custom balance reports.

### CODE REVIEW

**Comment by Xitian9** [comment]
I think the build error is not due to this PR.

File: hledger-lib/Hledger/Reports/MultiBalanceReport.hs [hunk 1]
```diff
@@ -25,6 +25,15 @@ module Hledger.Reports.MultiBalanceReport (
...
 
+  -- * Helper functions
+  calculateReportSpan,
+  makeReportQuery,
+  getPostingsByColumn,
+  getPostings,
+  calculateColSpans,
+  startingBalances,
+  generateMultiBalanceReport,
+
```
Comment (simonmichael) [inline hledger-lib/Hledger/Reports/MultiBalanceReport.hs:35]: These exports don't seem to be needed by hledger.. was it for some local code ?

File: hledger-lib/Hledger/Reports/ReportTypes.hs [hunk 2]
```diff
@@ -157,10 +152,11 @@ data CompoundPeriodicReport a b = CompoundPeriodicReport
 
 -- | Description of one subreport within a compound balance report.
 -- Part of a "CompoundBalanceCommandSpec", but also used in hledger-lib.
-data CBCSubreportSpec = CBCSubreportSpec
+data CBCSubreportSpec a = CBCSubreportSpec
   { cbcsubreporttitle          :: String
   , cbcsubreportquery          :: Journal -> Query
-  , cbcsubreportnormalsign     :: NormalSign
+  , cbcsubreportoptions        :: ReportOpts -> ReportOpts
+  , cbcsubreporttransform      :: PeriodicReport DisplayName MixedAmount -> PeriodicReport a MixedAmount
```
Comment (simonmichael) [inline hledger-lib/Hledger/Reports/ReportTypes.hs:159]: Would you mind adding brief explanatory haddocks for the fields here, at least the new cbcsubreportoptions and cbcsubreporttransform.

File: hledger-lib/Hledger/Reports/MultiBalanceReport.hs (hunk 1 above)
Comment (Xitian9) [reply hledger-lib/Hledger/Reports/MultiBalanceReport.hs:35]: Yes, if anybody wants to write their own report functions (which I have done occasionally), it's helpful to have access to them.

File: hledger-lib/Hledger/Reports/ReportTypes.hs (hunk 2 above)
Comment (Xitian9) [reply hledger-lib/Hledger/Reports/ReportTypes.hs]: Done.

**Comment by simonmichael** [comment]
Just a rebase needed now I think.

**Comment by Xitian9** [comment]
Rebased.

**Comment by simonmichael** [comment]
Thanks!

**Pull request merged**